entity-framework/core/modeling/value-comparers.md

```diff
@@ -0,0 +1,147 @@
+
+[!code-csharp[ConfigureImmutableStructProperty](../../../samples/core/Modeling/ValueConversions/MappingImmutableStructProperty.cs?name=ConfigureImmutableStructProperty)]
+
+EF Core has built-in support for generating compiled, memberwise comparisons of struct properties.
+This means structs don't need to have equality overridden for EF, but you may still choose to do this for [other reasons](/dotnet/csharp/programming-guide/statements-expressions-operators/how-to-define-value-equality-for-a-type).
+Also, special snapshotting is not needed since structs immutable and are always memberwise copied anyway.
+(This is also true for mutable structs, but [mutable structs should in general be avoided](/dotnet/csharp/write-safe-efficient-code).)
+
+### Mutable classes
+
+It is recommended that you use immutable types (classes or structs) with value converters when possible.
+This is usually more efficient and has cleaner semantics than using a mutable type.
+
+However, that being said, it is common to use properties of types that the application cannot change.
+For example, mapping a property containing a list of numbers: 
+
+[!code-csharp[ListProperty](../../../samples/core/Modeling/ValueConversions/MappingListProperty.cs?name=ListProperty)]
+
+The [`List<T>` class](/dotnet/api/system.collections.generic.list-1?view=netstandard-2.1):
+* Has reference equality; two lists containing the same values are treated as different.
+* Is mutable; values in the list can be added and removed.
+
+A typical value conversion on a list property might convert the list to and from JSON:
+
+[!code-csharp[ConfigureListProperty](../../../samples/core/Modeling/ValueConversions/MappingListProperty.cs?name=ConfigureListProperty)]
+
+This then requires setting a `ValueComparer<T>` on the property to force EF Core use correct comparisons with this conversion:
+
+[!code-csharp[ConfigureListPropertyComparer](../../../samples/core/Modeling/ValueConversions/MappingListProperty.cs?name=ConfigureListPropertyComparer)]
+
+> [!NOTE]  
+> The model builder ("fluent") API to set a value comparer has not yet been implemented.
+> Instead, the code above calls SetValueComparer on the lower-level IMutableProperty exposed by the builder as 'Metadata'.
+
+The `ValueComparer<T>` constructor accepts three expressions:
+* An expression for checking quality
+* An expression for generating a hash code
+* An expression to snapshot a value  
+
+In this case the comparison is done by checking if the sequences of numbers are the same.
+
+Likewise, the hash code is built from this same sequence.
+(Note that this is a hash code over mutable values and hence can [cause problems](https://ericlippert.com/2011/02/28/guidelines-and-rules-for-gethashcode/).
+Be immutable instead if you can.)
+
+The snapshot is created by cloning the list with ToList.
+Again, this is only needed if the lists are going to be mutated.
+Be immutable instead if you can. 
+
+> Point of interest:  
```

Comment:
Use [!NOTE] instead

Reply:
The guidance for NOTE is "This is a note which needs your attention, but it's not super important." In this case, this doesn't need anybody's attention--you don't need to know this.

But I'm on the fence. Do you still think NOTE is better?

Reply:
It renders much better and it's the most appropriate one from the in-built tags. Perhaps we can suggest that they add something like `!FYI`

Reply:
I'll use NOTE fore now.